workspace/solution.go

```diff
@@ -87,3 +83,37 @@ func (s *Solution) PathToParent() string {
 	}
 	return filepath.Join(dir, s.Track)
 }
+
+// SolutionMetadataFilepath is the path of the solution metadata file relative to the workspace.
+func SolutionMetadataFilepath() string {
+	return filepath.Join(ignoreSubdir, solutionFilename)
+}
+
+func createIgnoreSubdir(path string) error {
+	path = filepath.Join(path, ignoreSubdir)
+	if _, err := os.Stat(path); os.IsNotExist(err) {
+		if err := os.Mkdir(path, os.FileMode(0755)); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func migrateLegacySolutionFile(legacySolutionPath string, solutionPath string) error {
+	if _, err := os.Lstat(legacySolutionPath); err != nil {
+		return err
+	}
+	if err := createIgnoreSubdir(filepath.Dir(legacySolutionPath)); err != nil {
+		return err
+	}
+	if _, err := os.Lstat(solutionPath); err != nil {
+		if err := os.Rename(legacySolutionPath, solutionPath); err != nil {
+			return err
+		}
+		fmt.Fprintf(os.Stderr, "\nMigrated solution metadata to %s\n", solutionPath)
+	} else {
+		// TODO: decide how to handle case where both legacy and modern metadata files exist
+		fmt.Fprintf(os.Stderr, "\nAttempted to migrate solution metadata to %s but file already exists\n", solutionPath)
```

Comment:
I think the new metadata takes precendence which leads into my next comment about workspace,checkSolutionFile. It currently checks for a legacy metdata first, but I think we should check for the new path first and only check for legacy if it doesn’t exist.